python/tests/integration/arcticdb/test_vector_db.py

```diff
@@ -0,0 +1,229 @@
+import string
+
+from hypothesis import example, given, strategies as st
+from hypothesis.extra.pandas import column, data_frames, range_indexes
+from arcticdb.options import LibraryOptions
+from arcticdb_ext.version_store import NoSuchVersionException
+from arcticdb.util.hypothesis import (
+    numeric_type_strategies,
+    string_strategy,
+    use_of_function_scoped_fixtures_in_hypothesis_checked,
+)
+
+try:
+    from arcticdb.version_store import VersionedItem as PythonVersionedItem
+except ImportError:
+    # arcticdb squashes the packages
+    from arcticdb._store import VersionedItem as PythonVersionedItem
+from arcticdb.version_store.vector_db import VectorDB, new_vector_db
+
+import pytest
+import pandas as pd
+import numpy as np
+from arcticdb_ext.tools import AZURE_SUPPORT
+from arcticdb.util.test import assert_frame_equal
+import random
+
+if AZURE_SUPPORT:
+    pass
+
+try:
+    from arcticdb.version_store.library import (
+        WritePayload,
+        ArcticUnsupportedDataTypeException,
+        ReadRequest,
+        ReadInfoRequest,
+        ArcticInvalidApiUsageException,
+        StagedDataFinalizeMethod,
+    )
+except ImportError:
+    # arcticdb squashes the packages
+    from arcticdb.library import (
+        WritePayload,
+        ArcticUnsupportedDataTypeException,
+        ReadRequest,
+        ReadInfoRequest,
+        ArcticInvalidApiUsageException,
+        StagedDataFinalizeMethod,
+    )
+
+
+def test_top_k(arctic_client):
```

Comment:
This would be a good candidate for a hypothesis test. Hypothesis will generate random dataframes (see `test_filtering.py` for some examples). Then we can generate a random value for `k` that is less than the number of columns, and the rest of the test will stay the same. It will probably take a long time to run though, so will need to tune the maximum number of samples.